docs/digital-identity.md

```diff
@@ -0,0 +1,201 @@
+failed due to some error which wasn't reported.
+
+Check with them if the API call was made and if it was successful.  If
+the call was made but failed, then continue debugging account-api.
+
+### Can't log in
+
+Find out from the user what the email address they've used for their
+account is.
+
+You can find their user record with:
+
+```ruby
+user = OidcUser.find_by(email: "address")
+```
+
+#### Can't log in for the first time
+
+If there is no `OidcUser` record for that email address, this is the
+user's first time logging in.  When a user first logs in, we create
+their `OidcUser` record and fetch their email address.
+
+##### Check Sentry for `ActiveRecord::RecordNotUnique` events
+
+An exception is raised if two `OidcUsers` have the same subject
+identifier:
+
+```
+ActiveRecord::RecordNotUnique: PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "index_oidc_users_on_sub"`
+DETAIL:  Key (sub)=(foo) already exists.
+```
+
+This should never happen in production.
+
+As subject identifiers come from Digital Identity, contact them for
+support and give the duplicate subject identifier (`foo` in the
+example above).
+
+##### Check Sentry for `CapturedSensitiveException` events
+
+Try the debugging steps described above under "email address out of
+date": if there is another user record with the user's email address,
+creating their account will fail.
+
+##### Check Sentry for `UserDestroyed` events
+
+An exception is raised if a user logs into an account which Digital
+Identity has previously told us is deleted:
+
+```
+AccountSession::UserDestroyed (AccountSession::UserDestroyed):
+
+app/lib/account_session.rb:27:in `initialize'
+app/controllers/internal/authentication_controller.rb:22:in `new'
+app/controllers/internal/authentication_controller.rb:22:in `callback'
+```
+
+This is because we create a `Tombstone` record when a user is
+destroyed.  This record holds the subject identifier, and we use it to
+end any active sessions that user still has.
+
+If a user logs into an account that has a subject identifier with a
+corresponding `Tombstone` record, Digital Identity have re-used a
+subject identifier, or un-deleted the account.  We currently assume
+this does not happen.
+
+Removing the `Tombstone` record will allow the user to log in.  But if
+this is an account which has been un-deleted, any data we held on them
+will be lost.
+
+#### Can't log in at other times
+
+Check that Digital Identity is up and running, and ask if they are
+seeing an elevated rate of invalid authorization codes or access
+tokens from GOV.UK.
+
+If Digital Identity is experiencing a problem, we just have to wait
+for them to fix it.
+
+Otherwise, we have a bug in account-api, or perhaps elsewhere in the
+GOV.UK stack.  One thing to try would be to clear our cached copy of
+the Digital Identity OIDC configuration:
+
+```ruby
+Rails.cache.clear
+```
+
+Otherwise, there isn't any more specific guidance I can give here.
```

Comment:
```suggestion
```

Try to avoid use of first person in docs. Also not much value to this closing sentence.